### ISSUE TITLE
Scroll Button stops working

### ISSUE THREAD

**Issue by mrugeshrocks** [tcorreabr/mouse-swipe#5]
After enabling extension, swip gestures works perfectly fine.

But scrolling by middle button and middle button click stops working.

Tried to comment the middle button in config file but didnt workout.

Pls help

**Comment by tcorreabr** [comment]
Were you already a user and the error appeared in the latest release?

Can you paste the contents of your /etc/mouse-swipe.conf file?

**Comment by AL292** [comment]
I do have the more or less the same problem, because the click with the scroll wheel does work and paste on the shell or other place. Before, it was a question of changing the "path" or more precisely, the version of the python interpreter on shebang in main.py, as also in the systemd-unit mouse-swipe.service in the `[Service] ExecStart=python3.10 /usr/local/share/mouse-swipe/main.py`  , then I `sudo systemctl disable --now mouse-swipe.service` ; `sudo systemctl daemon-reload` and finally `sudo systemctl enable --now mouse-swipe.service` and all worked. I do have changed my mouse-swipe.conf as it follows , and always have worked.

```
[BTN_MIDDLE]
#    scroll = True
#    freeze = True

# Gnome shortcuts
[BTN_RIGHT]
    swipe_left = KEY_LEFTALT + KEY_LEFT
    swipe_right = KEY_LEFTALT + KEY_RIGHT
#    swipe_up = KEY_LEFTMETA
#    swipe_down = KEY_LEFTALT + KEY_TAB
```
I am in Zorin OS 17.3 which have python 3.10.12 more than 2 years working solid until this week that came an update that touched the python (I say this because when it was upgrading, my eyes catched python-samba and didn't have seen that package before), also I've tried in Debian 13 that ships with python3.13 that I have freshly installed today on another drive, and nothing, the results were the same, as far that you enabled the systemd-unit, or even sudo ./uninstall and then sudo ./install from the mouse-swipe folder, the wheel become unresponsive, but the swipe left and right do work.  
I have tried to refactor the code into C++ with the help of chatGPT, but it doesn't work like you have it working.. This is a tool so essencial in my working habits that I have spent hours trying to figure out how to keep it working. Only for reference, once I wanted to work with openSUSE Leap 15.6, but opensuse still have great part of their system depending on python2.7, and I had to give up because I have never could put mouse-swipe to work, even altinstall another version of python, the 3.8. I did the test that you mention on mouse-swipe.conf `python3 -m evdev.evtest` and the mouse buttons are recognized, namely the right one 273.

I hope have given enough context for you, my friend, try to solve this please. Did I told you that I hate Python?? 
YES I DO HATE THAT THING. 
 I've spent hours plus hours trying to debug this to no avail, of course, other learnings and works, became on hold.

edit: some typos

**Comment by AL292** [comment]
But here I am again, and after re-reading the @mrugeshrocks comment, I went to do the only thing that I have not done..
I commented the reference to the [BTN_MIDDLE] so 

my  mouse-swipe.conf looks like 
```
# Options for mouse buttons:
#   - swipe_left, swipe_right, swipe_up, swipe_down: Mappings for mouse swipes.
#   - scroll: Button will support vertical and horizontal scroll when pressed.
#   - freeze: Mouse cursor will not move while button is pressed.
#   - click: Mapping for mouse click (without movement). If not defined, the default click action will be performed.
#
# Notes:
#   - If using scroll option, the swipe_* options are ignored for that button.
#   - For buttons and key names (BTN_* and KEY_*), run: python -m evdev.evtest.

#[BTN_MIDDLE]
#    scroll = True
#    freeze = True

# Gnome shortcuts
[BTN_RIGHT]
    swipe_left = KEY_LEFTALT + KEY_LEFT
    swipe_right = KEY_LEFTALT + KEY_RIGHT
#    swipe_up = KEY_LEFTMETA
#    swipe_down = KEY_LEFTALT + KEY_TAB
```

my main.py looks like 
```
#!/usr/bin/python3.10

import asyncio, copy, logging
from evdev import InputDevice, ecodes, list_devices
from systemd.journal import JournalHandler
from virtual_device import create_virtual_device, remove_virtual_device
from config import read_config_file
from mouse import Mouse

def get_mouses():
    logger.info("Searching for mouses..")

    global number_of_devices

(....)
```

and my mouse-swipe.service looks like 
```
[Unit]
Description=Swipe gestures support for mouse buttons
After=display-manager.service

[Service]
ExecStart=python3.10 /usr/local/share/mouse-swipe/main.py
RemainAfterExit=no
Restart=on-failure
RestartSec=5s

[Install]
WantedBy=multi-user.target
```

And `sudo ./uninstall` and `sudo ./install` and..... IT WORKS AGAIN!!!!!  Did I ever mention that I hate Python?

**Comment by tcorreabr** [comment]
The latest commits are precisely about the scroll wheel. Maybe I introduced a bug with them. Can you test if the problem persists if you use the commit before them? (like [f3e72e2](https://github.com/tcorreabr/mouse-swipe/commit/f3e72e281bb8d637db8681666d88b6769f1c7559))

**Comment by AL292** [comment]
Yes, I've tried until 4 versions before this last one, and the problem was still there, what is odd because I had this setup saved in a folder separated from the git clone folder which is your repo. 
I don't know what changed in the python interpreter bundle, but as soon as I commented that line, it started to work again.

**Comment by AL292** [comment]
As the time went by, the problem still persisted. Today, when I was trying once again to figure it out what was wrong, I was editing the files mouse-swipe.conf and main.py on Kate editor that I have installed even I am using Zorin based on Gnome, but I like pretty much the KDE tools, and the Kate editor warned me about the missing LSP Client located at [https://github.com/python-lsp/python-lsp-server](url) and I went there and did the install like they say `sudo apt-get install python3-pylsp` then the output of my shell was the posted bellow, even it is in Portuguese, it's easy to understand what more python related packages it needed
```shell
sudo apt-get install python3-pylsp
A ler as listas de pacotes... Pronto
A construir árvore de dependências... Pronto
A ler a informação de estado... Pronto      
Os pacotes adicionais seguintes serão instalados:
  python3-jedi python3-parso python3-pluggy python3-pylsp-jsonrpc python3-ujson
Pacotes sugeridos:
  flake8 pylint python3-autopep8 python3-mccabe python3-pycodestyle python3-pydocstyle python3-pyflakes python3-rope
  python3-yapf
Serão instalados os seguintes NOVOS pacotes:
  python3-jedi python3-parso python3-pluggy python3-pylsp python3-pylsp-jsonrpc python3-ujson
``` 
Now it all works again. Hopefuly forever.  I think I haven't told  you, but I hate Python code.

**Comment by tcorreabr** [comment]
Strange. I don't have the equivalent of that package installed on my distro (python-lsp-server on arch), and yet everything works normally here.

I'm not a big fan of Python either. I developed this project because I was trying to learn the language.

(A propósito, eu sou brasileiro então até que consigo me virar bem com a língua portuguesa 😀)

**Comment by AL292** [comment]
Well. It was good until it lasted.. I needed to reboot my machine, and the problem appeared again. the swipe left and right with the right button of the mouse work as it always have worked, but the scroll wheel loose completly it's function, unless you press the scroll wheel and move the mouse up or down, and the webpages go up or down, but the paste with the middle button (scroll wheel) function doesn't function anymore. 

Yesterday, after the installation of those modules that I've mentioned in the comment before, I did 3 reboot cycles to guarantee that what I would be writting it was flawless.  When I went to sleep, I put the laptop to sleep, and today I had a recurring problem when the system gets out of sleep, that the pulseaudio.service became inactive (dead) and for a strange reason, the `systemctl --user restart  pulseaudio.service`  didn't brought back the audio, hence the need to reboot the machine... and here we are again.. I didn't want to but probably I need to do a fresh install of an OS, and loose years of customizations I've made so far, and the workflow  that they brought to my day-to-day life.

But I think that (if you have the time and knowledge) this tool written in golang would be awesome.

Thank you for making this amazing tool.
